R/string_operations.r

```diff
@@ -212,6 +214,14 @@ col_strsplit <- function(x, split, ...) {
   plain <- strip_style(x)
   splits <- re_table(split, plain, ...)
   chunks <- non_matching(splits, plain, empty = TRUE)
+  # drop any zero length trailing chunks that are created when the split
+  # happens at end of string; this is to replicate 'substr' behavior
+  chunks <- lapply(
+    chunks,
+    function(y)
+      if(nrow(y) > 1L && !tail(y, 1L)[, "length"]) head(y, -1L) else y
```

Comment:
I'll have to look at it again later.  Clearly this is wrong, but I think I was trying to ensure that we don't drop this:

```
> strsplit("a", "a")[[1]]
[1] ""
```
Hopefully I can look at it this evening.